### ISSUE TITLE
Invalid `no-loop-func` error

### ISSUE THREAD

**Issue by MayhemYDG** [eslint/eslint#5133]
>The version of ESLint you are using.

1.10.3 or http://eslint.org/demo/

>What you did.

Disabled every rules except for `no-loop-func`, and enabled the `es6` environment.

```js
const obj = {
	a: [1, 2]
};

for (const key in obj) {
	obj[key].forEach(val => t(val));
}
```

>What you expected to happen.

No `no-loop-func` error.

>What actually happened.

`6:19 - Don't make functions within a loop (no-loop-func)`

Surprisingly, if you replace
`obj[key].forEach(val => t(val));`
with 
`obj[key].forEach(val => val);`
you don't get that error anymore.

**Comment by eslintbot** [comment]
@MayhemYDG Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
This has been fixed in #5044 just a few days ago, it should be released as part of 2.0.0. Closing, as this is already done. Feel free to reopen if you run into this problem again after 2.0.0 released.
